src/composables/incomingTransactions.ts

```diff
@@ -0,0 +1,74 @@
+import { ref } from '@vue/composition-api';
+import VueRouter from 'vue-router';
+import { WEB_SOCKET_CHANNEL_TYPES, roundAmountToPrecision } from '../popup/utils';
+import { i18n } from '../store/plugins/languages';
+import { IDefaultComposableOptions, ITransaction } from '../types';
+import { useAccounts } from './accounts';
+import { useNotifications } from './notifications';
+import { useGetter } from './vuex';
+import { useWebSocket } from './websocket';
+import { ROUTE_TX_DETAILS } from '../popup/router/routeNames';
+
+export interface IIncomingTransactionsOptions extends IDefaultComposableOptions {
+  router: VueRouter
+}
+
+export function useIncomingTransactions({ store, router }: IIncomingTransactionsOptions) {
+  const { watchChannelUpdates } = useWebSocket({ store });
+  const { addWalletNotification } = useNotifications({ store });
+  const { isLocalAccountAddress, getAccountNameByAddress } = useAccounts({ store });
+  const getTxAmountTotal = useGetter<any>('getTxAmountTotal');
+  const getTxSymbol = useGetter<any>('getTxSymbol');
+  const unWatchChannelUpdates = ref();
+
+  /**
+   * This function starts listening for incoming transactions on the 'Transactions' channel.
+   */
+  function startListeningForIncomingTransactions() {
+    const receivedTransactions: Set<string> = new Set<string>();
```

Comment:
No need to set the type twice.
```suggestion
    const receivedTransactions = new Set<string>();
```